app/src/main/java/io/github/cmput301f19t19/legendary_fiesta/ui/OwnMoodsFragment.java

```diff
@@ -22,36 +23,49 @@
     private Activity mActivity; //reference to associated activity class, initialized in onAttach function
     private View mView; //reference to associated view, initialized in onCreateView
 
+    Spinner filterSpinner;
+    String selectedFilter;
+
     public View onCreateView(@NonNull LayoutInflater inflater,
                              ViewGroup container, Bundle savedInstanceState) {
 
         View root = inflater.inflate(R.layout.fragment_own_moods, container, false);
         mView = root;
 
-        setUpSpinner();
+        setUpFilterSpinner();
         return root;
     }
 
     @Override
     public void onAttach(Context context){
         super.onAttach(context);
-        /**
-         * get reference to associated activity
-         */
+        // get reference to associated activity
         mActivity = (Activity) context;
     }
 
     /**
      * Mostly a test function to set up the spinner, populate it with a string array from resource.xml
      */
-    private void setUpSpinner(){
-        Spinner spinner = mView.findViewById(R.id.filter_spinner);
+    private void setUpFilterSpinner(){
+        filterSpinner = mView.findViewById(R.id.filter_spinner);
 
         ArrayAdapter<CharSequence> spinnerAdapter = ArrayAdapter.createFromResource(mActivity, R.array.filter,
```

Comment:
There's on error when clicking on the `Filter` bar.
Returns `java.lang.IllegalStateException: ArrayAdapter requires the resource ID to be a TextView`

Reply:
Outdated implementation, resolved.